### PR TITLE
Migrate 'LegendCategories' Component from makeStyles to styled-component + cleanup

### PR DESCRIPTION
**Pull request by jantolg**
# Description

This PR includes a refactorization of LegendCategories component, all classes from makestyles are been converted to styled components

Shortcut:  [297037](https://app.shortcut.com/cartoteam/story/297037/migrate-away-from-makestyles-i-widgets)

## Type of change

- Refactor

# Acceptance

The code resultant must return same styles component visualization as the same before changes

1. Overview a general visualization from v1.x to v2.x
2. Compare visualization from v1.x to v2.x with prop isMax setted and unsetted
3. Compare visualization from v1.x to v2.x with prop isStrokeColor setted and unsetted
4. Compare visualization from v1.x to v2.x with prop maskedIcon setted and unsetted
5. Compare visualization from v1.x to v2.x with prop icon setted and unsetted
5. Compare visualization from v1.x to v2.x with prop color setted and unsetted

# Basic checklist

- Good PR name
- Shortcut link
- Changelog entry
- Just one issue per PR
- GitHub labels
- Proper status & reviewers
- Tests
- Documentation

### CODE REVIEW

File: packages/react-ui/src/widgets/legend/LegendCategories.js [hunk 1]
```diff
@@ -65,61 +64,70 @@ LegendCategories.propTypes = {
...
-    width: '12px',
-    height: '12px',
+const getCircleStyles = ({ isMax, color, isStrokeColor, theme }) => ({
+  border: '2px solid transparent',
+  '&::after': {
+    position: 'absolute',
+    display: isMax ? 'block' : 'none',
+    content: '""',
+    width: '16px',
+    height: '16px',
```
Comment (vmilan) [inline packages/react-ui/src/widgets/legend/LegendCategories.js:74]: Please, use `theme.spacing()` for size values like this

File: packages/react-ui/src/widgets/legend/LegendCategories.js [hunk 2]
```diff
@@ -65,61 +64,70 @@ LegendCategories.propTypes = {
...
+});
+
+const Marker = styled(Box, {
+  shouldForwardProp: (prop) =>
+    !['isMax', 'icon', 'maskedIcon', 'color', 'isStrokeColor'].includes(prop)
+})(({ isMax, icon, maskedIcon, color, isStrokeColor, theme }) => ({
+  whiteSpace: 'nowrap',
+  display: 'block',
+  width: '12px',
+  height: '12px',
```
Comment (vmilan) [inline packages/react-ui/src/widgets/legend/LegendCategories.js:107]: Please, use `theme.spacing()` for size values like this

**Review by vmilan — APPROVED** [review]
LGTM. Just some minor details

File: packages/react-ui/src/widgets/legend/LegendCategories.js (hunk 1 above)
Comment (VictorVelarde) [reply packages/react-ui/src/widgets/legend/LegendCategories.js:74]: Agree, and then we can merge

**Review by VictorVelarde — APPROVED** [review]
👍🏻

**Comment by VictorVelarde** [comment]
Coveralls service is broken, but build & tests are ok

**Pull request merged**